src/components/tree-item/tree-item.tsx

```diff
@@ -0,0 +1,291 @@
+    this.numberOfParentTrees = this.getParents(this.el);
+    //Get the item height
+    this.calculateItemHeight();
+    //If parent has checkbox...
+    const parentHasCheckbox = this.el.parentElement.getAttribute("checkbox");
+    if (parentHasCheckbox !== null) {
+      this.parentHasCheckbox = true;
+    }
+
+    //Defines the height of the vertical line that associates the chid items with the parent item
+    setTimeout(
+      function() {
+        this.calculateVerticalLineHeight();
+      }.bind(this),
+      500
+    );
+  }
+
+  @Watch("downloading")
+  downloadingHandler() {
+    if (this.downloading) {
+      this.hidePlusMinusIcon = true;
+      setTimeout(
+        function() {
+          this.showDownloadingIcon = true;
+        }.bind(this),
+        500
+      );
+    } else {
+      this.showDownloadingIcon = false;
+      setTimeout(
+        function() {
+          this.hidePlusMinusIcon = false;
+        }.bind(this),
+        500
+      );
+    }
+  }
+
+  getParents(elem) {
+    // Returns the number of parent tree items in order to set the appropriate paddding-left
+    // Set up a parent array
+    const numberOfTreeParents = [];
+
+    // Push each parent element to the array
+    for (; elem && elem !== document; elem = elem.parentNode) {
+      const elemTagNAme = elem.tagName;
+      if (elemTagNAme === "GXG-TREE") {
+        numberOfTreeParents.push(elem);
+      }
+    }
+    return numberOfTreeParents.length;
+  }
+
+  toggleTreeIconClicked() {
+    //If tree is collapsed, uncollapse.
+    //If tree is uncollapsed, collapse.
+    if (!this.isLeaf) {
+      if (this.open) {
+        this.open = false;
+      } else {
+        this.open = true;
+        //Play click open sound
+        const audio = new Audio(
+          getAssetPath("./tree-item-assets/click-open.mp3")
+        );
+        audio.play();
+      }
+      this.calculateItemHeight();
+      //Recalculate item`s parents tree-items height
+      this.itemToggled.emit();
+    }
+  }
+
+  liTextDoubleClicked() {
+    this.toggleTreeIconClicked();
+  }
+
+  returnToggleIconType() {
+    //Returns the type of icon : gemini-tools/add or gemini-tools/minus
+    if (this.open) {
+      return "gemini-tools/minus";
+    } else {
+      return "gemini-tools/add";
+    }
+  }
+
+  returnPaddingLeft() {
+    //returns the appropriate padding left to the .li-text element
+    let paddingLeft = 0;
+
+    paddingLeft = this.numberOfParentTrees * 35;
+    if (!this.isLeaf) {
+      paddingLeft -= 5;
+    }
+    this.itemPaddingLeft = paddingLeft;
+    return paddingLeft + "px";
+  }
+
+  calculateItemHeight() {
+    setTimeout(
+      function() {
+        this.height = this.el.offsetHeight;
+      }.bind(this),
+      300
+    );
+  }
+
+  //Lines
+  calculateVerticalLineHeight() {
+    //Returns the height of the vertical line that associates the chid-items with the parent item
+    this.verticalLineHeight = this.height - 29 + "px";
```

Comment:
I feel this ["magic" number](https://en.wikipedia.org/wiki/Magic_number_(programming)) could cause problems in the future.  How does it relate to the tokens defined in the CSS?

Reply:
This code was removed because I found a better way of calculating this value.